### PR TITLE
Add missed pinning dependencies

### LINKED ISSUE #259: Docs build is failing
## 🐛 Bug

We're getting a 

```
Extension error:
Could not import extension sphinx_tags (exception: cannot import name 'get_matching_files' from 'sphinx.util.matching' (/home/circleci/project/venv/lib/python3.10/site-packages/sphinx/util/matching.py))
make: *** [Makefile:30: docs-build] Error 2
```

See, for example, https://app.circleci.com/pipelines/github/napari/docs/365/workflows/73b5e766-7082-4c2f-9774-dffa61a364e6/jobs/355?invite=true#step-107-2326_16

I will try to take a look.

### PR DESCRIPTION
**Pull request by Czaki**
# Description
Closes #259
using the constraints file pins back sphinx_tags to 0.2.1 which is compatible with sphinx<5

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -52,7 +52,7 @@ jobs:
           name: Install python dependencies
           command: |
             . venv/bin/activate
-            python -m pip install -r requirements.txt
+            python -m pip install -r requirements.txt -c "napari/resources/constraints/constraints_py3.10_docs.txt"
```
Comment (melissawm) [inline .circleci/config.yml:55]: We need to update this constraint first, though: https://github.com/napari/napari/blob/467d57b3967074370b2ad6750bdfc1af8aee67d6/resources/constraints/constraints_py3.10_docs.txt#L507

Comment (Czaki) [reply .circleci/config.yml:55]: Why?

Comment (psobolewskiPhD) [reply .circleci/config.yml:55]: The import failure in https://github.com/napari/docs/issues/259 is due to sphinx < 5
(That import requires sphinx >5.1)
https://github.com/napari/docs/pull/256#issuecomment-1786315010

Comment (psobolewskiPhD) [reply .circleci/config.yml:55]: Oh, nvm, I see. The constraints pins back sphinx_tags to 0.2.1
https://github.com/napari/napari/blob/467d57b3967074370b2ad6750bdfc1af8aee67d6/resources/constraints/constraints_py3.10_docs.txt#L539

Comment (Czaki) [reply .circleci/config.yml:55]: Yes. constraints pins `sphinx-tags` and `napari-sphinx-theme`. 

And until update constraints files there will be no problem.  

Of course, relax sphinx in requirements.txt is required for constraints compilation provide newer sphinx and `napari-sphinx-theme`

**Review by melissawm — APPROVED** [review]
Sounds good, then!

**Review by psobolewskiPhD — APPROVED** [review]

**Comment by psobolewskiPhD** [comment]
@Czaki Can't suggest, but don't we need the constraint to be set in the Build phase too?
Line 60
```
environment:
            PIP_CONSTRAINTS: ../resources/constraints/constraints_py3.10_docs.txt
```

Same in the build_docs, but like like 51/52

**Comment by psobolewskiPhD** [comment]
Merging to get docs to build

**Pull request merged**